### PR TITLE
Allow sync IO for NewtonsoftJsonOutputFormatter

### PR DESCRIPTION
**Pull request by Tratcher**
#8302 This was missed before because there were no tests that wrote a large enough body to trigger a sync flush.

### CODE REVIEW

**Comment by Tratcher** [comment]
@rynowak I'm going dark soon. If I don't get this in today can you take over?

File: src/Mvc/test/Mvc.FunctionalTests/JsonOutputFormatterTests.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public async Task JsonOutputFormatter_ReturnsIndentedJson()
             {
                 Id = 1,
                 Alias = "john",
-                description = "Administrator",
+                description = "This is long so we can test large objects " + new string('a', 1024 * 65),
```
Comment (rynowak) [inline src/Mvc/test/Mvc.FunctionalTests/JsonOutputFormatterTests.cs]: <3

**Review by rynowak — APPROVED** [review]

**Comment by rynowak** [comment]
Looks like you have some nulls in unit tests. I suspect it's some **ancient** tests that mock the http context instead of newing it up.

**Comment by Tratcher** [comment]
Ack.

**Comment by rynowak** [comment]
> @rynowak I'm going dark soon. If I don't get this in today can you take over?

sure

**Comment by ikirkpat** [comment]
We are using dotnet 3.1 and seem to be having crashes at serialization time. I'm investigating whether or not [8302](https://github.com/dotnet/aspnetcore/issues/8302) is the reason for this. It seems that this PR implies that async IO is not working properly for large responses so this PR just made a workaround to let you serialize syncronously. Is that correct? Has this ever fixed so we can do async IO for JSON? If so, what version of dotnet has the fix?

**Pull request merged**